### PR TITLE
Merge/improve regret operators

### PR DESCRIPTION
**Pull request by hkgrov**

### CODE REVIEW

File: heuristic/delta_calculations.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def calculate_f_for_employee(L_C_D, days, e, saturdays, state, weeks, weights):
 
     f = (
         sum(
-            weights["rest"] * min(100, state.w[e, j][1])
+            weights["rest"] * min(72, state.w[e, j][1])
```
Comment (ekallevik) [inline heuristic/delta_calculations.py:121]: Use `WEEKLY_REST_HOURS[1]` here

File: heuristic/delta_calculations.py [hunk 2]
```diff
@@ -263,20 +263,20 @@ def calc_weekly_objective_function(state, competencies, time_periods_in_week, co
 
         if setting == "worst":
             value[j] = (
-                        sum(min(100, state.w[e, j][1]) for e in employees)
+                        sum(min(72, state.w[e, j][1]) for e in employees)
```
Comment (ekallevik) [inline heuristic/delta_calculations.py:266]: Use `WEEKLY_REST_HOURS[1]` here

File: heuristic/delta_calculations.py [hunk 3]
```diff
@@ -442,30 +442,31 @@ def regret_objective_function(state, employee, off_shifts, saturdays, days, L_C_
...
+
 def regret_weekly_rest(state, shifts_at_week, e, week, shift):
     actual_shifts = [(t, v) for t,v in shifts_at_week[week] if state.x[e,t,v] == 1 or (t,v) == (shift[0],shift[1])]
     off_shift_periods = []
     week_interval = [7 * 24 * week, 7 * 24 * (week + 1) ]
   
     if not actual_shifts:
         off_shift_periods.append(float(week_interval[1] - week_interval[0]))
-        return min(50, off_shift_periods[0])
+        return 0.5 * min(72, off_shift_periods[0])
```
Comment (ekallevik) [inline heuristic/delta_calculations.py:469]: Use `WEEKLY_REST_HOURS[1]`

File: heuristic/delta_calculations.py [hunk 4]
```diff
@@ -480,29 +481,28 @@ def regret_weekly_rest(state, shifts_at_week, e, week, shift):
                 off_shift_periods.append(actual_shifts[i+1][0] - (actual_shifts[i][0] + actual_shifts[i][1]))
 
     if off_shift_periods:
-        return min(50, max(off_shift_periods))
-    #Positive if we have a rest period. I use 50 as return if there is since we use 0.5 in the weights
+        return 0.5 * min(72, max(off_shift_periods))
```
Comment (ekallevik) [inline heuristic/delta_calculations.py:484]: `WEEKLY_REST_HOURS[1]`

File: heuristic/heuristic_calculations.py [hunk 5]
```diff
@@ -33,7 +33,9 @@ def calculate_weekly_rest(data, x, w):
 
             if actual_shifts[key][0][0] - important[week] >= 36:
                 off_shift_periods[key].append((important[week], actual_shifts[key][0][0] - important[week]))
-
+                
+    for key in off_shift_periods:
+            w[key] = max(off_shift_periods[key],key=itemgetter(1))
```
Comment (ekallevik) [inline heuristic/heuristic_calculations.py:38]: Use 
```python
min(
    WEEKLY_REST_HOURS[1],
    max(off_shift_periods[key],key=itemgetter(1))
```

here?

File: heuristic/heuristic_calculations.py [hunk 6]
```diff
@@ -139,7 +141,7 @@ def calculate_f_for_employee(data, e, soft_vars, weights, w):
 
     f = (
         weights["rest"] * sum(
-            min(100, w[e, j][1])
+            min(72, w[e, j][1])
```
Comment (ekallevik) [inline heuristic/heuristic_calculations.py:144]: Remove the min-calculation here if you implement my previous suggestion

File: heuristic/heuristic_calculations.py (hunk 5 above)
Comment (hkgrov) [reply heuristic/heuristic_calculations.py:38]: I think I will continue to use 
'''
w[key] = max(off_shift_periods[key],key=itemgetter(1))
'''
It makes more sense to actually give assign the employee to the rest period even though they are not rewarded for it. Or else we would just cut off rest periods and that would seem weird in the visualisation of the schedules. The restriction on reward should be maintained in the objective value calculations

File: heuristic/heuristic_calculations.py (hunk 6 above)
Comment (hkgrov) [reply heuristic/heuristic_calculations.py:144]: See my comment above. I believe this is a better way to do it. We should not restrict an employees rest period if they are able to have longer breaks. We should just not reward the longer periods

File: heuristic/heuristic_calculations.py (hunk 5 above)
Comment (ekallevik) [reply heuristic/heuristic_calculations.py:38]: Fair point!

**Pull request merged**